lib/utils/desktop_drop_downloader.dart

```diff
@@ -0,0 +1,48 @@
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+
+import 'package:http/http.dart';
+import 'package:path_provider/path_provider.dart';
+
+abstract class DesktopDropDownloader {
+  const DesktopDropDownloader._();
+
+  static Future<String?> unsupportedUriCallback(String url) async {
+    if (kIsWeb) return null;
+    final uri = Uri.tryParse(url);
+    if (uri == null) return null;
+
+    if (!['http', 'https'].contains(uri.scheme)) return null;
+
+    Response response;
+
+    try {
+      response = await get(uri);
```

Comment:
This seems to have no user feedback. When I move an url to a 100GB file into it, will it just try to download 100GB into RAM without the GUI showing any feedback?